src/pages/api/dogs/[id].js

```diff
@@ -51,6 +53,27 @@ export default async function handler(req, res) {
         return;
       }
 
+      const token = await getToken({ req });
```

Comment:
use `const session = await getServerSession(req, res, authOptions);` for consistency